pkg/rpc/client/helper.go

```diff
@@ -168,16 +168,20 @@ func createIteratorUnwrapperScript(contract util.Uint160, operation string, para
 }
 
 // topIterableFromStack returns the list of elements of `resultItemType` type from the top element
-// of the provided stack. The top element is expected to be an Array, otherwise an error is returned.
-func topIterableFromStack(st []stackitem.Item, resultItemType interface{}) ([]interface{}, error) {
+// of the provided stack. The top element is expected to be an Array or InteropInterface with Iterator
+// inside, otherwise an error is returned. If the top element is Iterator with iterator session ID, then
+// first DefaultMaxIteratorResultItems elements will be retrieved via single `traverseiterator` JSON-RPC call.
+func (c *Client) topIterableFromStack(res *result.Invoke, resultItemType interface{}) ([]interface{}, error) {
+	st := res.Stack
 	index := len(st) - 1 // top stack element is the last in the array
-	if t := st[index].Type(); t != stackitem.ArrayT {
-		return nil, fmt.Errorf("invalid return stackitem type: %s (Array expected)", t.String())
+	if st[index].Type() != stackitem.ArrayT {
```

Comment:
These hunks probably belong to earilier commits, they do nothing useful here.

Reply:
Yes, fixed.